lib/webpacker/dev_server.rb

```diff
@@ -0,0 +1,73 @@
+# Same convention as manifest/configuration.rb
+
+# Loads webpacker configuration from config/webpacker.yml
+
+require "webpacker/configuration"
+
+class Webpacker::DevServer < Webpacker::FileLoader
+  class << self
+    def enabled?
+      case ENV["WEBPACKER_DEV_SERVER"]
+      when /true/i then true
+      when /false/i then false
+      else !data[:dev_server].nil?
+      end
+    end
+
+    # read settings for dev_server
+    def hot_reloading?
+      if enabled?
+        case ENV["WEBPACKER_HMR"]
+        when /true/i then true
+        when /false/i then false
+        else fetch(:hot_reloading)
+        end
+      else
+        false
```

Comment:
The current idea is that the hot reloading setting only applies if the dev server is enabled.

An alternative interpretation could be that the ENV for hot reloading overrides the dev server setting. I think that's getting too complicated.